end-to-end-tests/src/bin/bootstrap.rs

```diff
@@ -0,0 +1,100 @@
+use anyhow::{bail, Result};
+use end_to_end_tests::helpers::ctx::{build_client, nexus_addr, Context};
+use end_to_end_tests::helpers::generate_name;
+use omicron_test_utils::dev::poll::{wait_for_condition, CondCheckError};
+use oxide_client::types::{
+    ByteCount, DiskCreate, DiskSource, IpPoolCreate, IpRange, Ipv4Range,
+};
+use oxide_client::{ClientDisksExt, ClientIpPoolsExt, ClientOrganizationsExt};
+use std::net::IpAddr;
+use std::time::Duration;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let client = build_client()?;
+
+    // ===== ENSURE NEXUS IS UP ===== //
+    eprintln!("waiting for nexus to come up...");
+    wait_for_condition(
+        || async {
+            client
+                .organization_list()
+                .send()
+                .await
+                .map_err(|_| CondCheckError::<oxide_client::Error>::NotYet)
+        },
+        &Duration::from_secs(1),
+        &Duration::from_secs(300),
+    )
+    .await?;
+
+    // ===== CREATE IP POOL ===== //
+    eprintln!("creating IP pool...");
+    let nexus_addr = match nexus_addr().ip() {
+        IpAddr::V4(addr) => addr.octets(),
+        IpAddr::V6(_) => bail!("not sure what to do about IPv6 here"),
+    };
+    // TODO: not really sure about a good heuristic for selecting an IP address
+    // range here. in both my (iliana's) environment and the lab, the last octet
+    // is 20; in my environment the DHCP range is 100-249, and in the buildomat
+    // lab environment the network is currently private.
```

Comment:
Carving some space out of the same subnet as Nexus seems alright for now. Besides Nexus (and the global zone) was there anything else we'd conflict with?

Reply:
I'm mostly worried about home network setups for people running these tests locally. I don't think there's a whole lot I can do there though.

(I've idly been wondering if we should have an out-of-repo config we look for describing a user's network setup.)

Reply:
> (I've idly been wondering if we should have an out-of-repo config we look for describing a user's network setup.)

Yes! I think that or even just something we stick in the .gitignore. Since the configs are part the repo today any modifications you make for your local testing (e.g. mac gateway) pollute git status.